test/serializer/optimized-functions/OptimizeInSpeculativeContext.js

```diff
@@ -0,0 +1,37 @@
+let obj1 = global.__abstract
+  ? __abstract("object", '({get foo() { return "bar"; }})')
+  : {
+      get foo() {
+        return "bar";
+      },
+    };
+let obj2 = global.__abstract ? __abstract("object", '({foo:{bar:"baz"}})') : { foo: { bar: "baz" } };
+if (global.__makeSimple) {
+  __makeSimple(obj2);
+}
+
+function additional1() {
+  function foo() { return 2; }
+  if (global.__optimize) __optimize(foo);
+  global.foo = foo;
+  return String(obj1.foo);
+}
+
+function additional2() {
+  function bar() { return 5; }
```

Comment:
I don't see any assertion that this inner function is actually getting optimized?